app/utils/file/file_picker/index.ts

```diff
@@ -0,0 +1,331 @@
+                ),
+                text: formatMessage({
+                    id: 'mobile.storage_permission_denied_description',
+                    defaultMessage:
+                        'Upload files to your Mattermost instance. Open Settings to grant Mattermost Read and Write access to files on this device.',
+                }),
+            },
+            denied_ios: {
+                title: formatMessage(
+                    {
+                        id: 'mobile.ios.photos_permission_denied_title',
+                        defaultMessage:
+                            '{applicationName} would like to access your photos',
+                    },
+                    {applicationName},
+                ),
+                text: formatMessage({
+                    id: 'mobile.ios.photos_permission_denied_description',
+                    defaultMessage:
+                        'Upload photos and videos to your Mattermost instance or save them to your device. Open Settings to grant Mattermost Read and Write access to your photo and video library.',
+                }),
+            },
+            denied_android: {
+                title: formatMessage(
+                    {
+                        id: 'mobile.android.photos_permission_denied_title',
+                        defaultMessage:
+                            '{applicationName} would like to access your photos',
+                    },
+                    {applicationName},
+                ),
+                text: formatMessage({
+                    id: 'mobile.android.photos_permission_denied_description',
+                    defaultMessage:
+                        'Upload photos to your Mattermost instance or save them to your device. Open Settings to grant Mattermost Read and Write access to your photo library.',
+                }),
+            },
+        };
+
+        return permissions[source];
+    };
+
+    private prepareFileUpload = async (files: Array<Asset | DocumentPickerResponse>) => {
+        const out = await extractFileInfos(files);
+
+        if (out.length > 0) {
+            DeviceEventEmitter.emit(Navigation.NAVIGATION_CLOSE_MODAL);
+            this.uploadFiles(out);
+        }
+    };
+
+    private getPermissionDeniedMessage = (source?: PermissionSource) => {
+        const sources = ['camera', 'storage', 'photo'];
+        const deniedSource: PermissionSource = Platform.select({android: 'denied_android', ios: 'denied_ios'})!;
+        const msgForSource = source && sources.includes(source) ? source : deniedSource;
+
+        return this.getPermissionMessages(msgForSource);
+    };
+
+    private getFilesFromResponse = async (response: ImagePickerResponse): Promise<Asset[]> => {
+        if (!response?.assets?.length) {
+            return [];
+        }
+
+        const files: Asset[] = [];
+
+        await Promise.all((response.assets.map(async (file) => {
+            if (Platform.OS === 'android') {
```

Comment:
We can reverse this if-else as:
```
if (Platform.OS === 'ios') {
   files.push(file)
   return
}
...
```
This way we remove a level of indentation.

Reply:
Updated - to using if else - more readable